internal/armada/server/submit.go

```diff
@@ -897,8 +897,23 @@ func (server *SubmitServer) applyDefaultsToPodSpec(spec *v1.PodSpec) {
 	if shouldDefaultPriorityClass {
 		spec.PriorityClassName = defaultPriorityClass
 	}
+
+	// add missing TerminationGracePeriod if needed
+	server.applyTerminationGracePeriodDefault(spec)
 }
 
+// applyTerminationGracePeriodDefault will give the podspec a default if needed
+func (server *SubmitServer) applyTerminationGracePeriodDefault(spec *v1.PodSpec) {
+	terminationGracePeriod := int64(server.schedulingConfig.TerminationGracePeriod.Default.Seconds())
+	preememptionEnabled := server.schedulingConfig.Preemption.Enabled
+	if preememptionEnabled &&
+		spec.PriorityClassName != "" &&
+		spec.TerminationGracePeriodSeconds == nil &&
+		terminationGracePeriod > 0 {
```

Comment:
Should we check if `terminationGracePeriod` is bounded by the max and min? I guess this turns into the question if we should validate that the default is in between the max and min?

Reply:
I guess this would be caught anyway by `validateTerminationGracePeriod` but it might still be clearer to validate the default here so the user realizes the problem lies with the server configuration and not their own pod-spec.

Reply:
I don't think it's an unreasonable thing to do. Do we validate the server config elsewhere? I'm still new to most of this codebase.

Reply:
I think we have to trust the configurer to put sane values in -- the whole system is easily broken by incorrect config and this is the least of those problems :-)

Reply:
I went ahead and added a separate error message to the validation method when the default is outside the boundaries